### PR TITLE
add_value method

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Description

Adds a add value method to loop over a list that should be mapped to the same mode and sum them for instances where multiple source type maps to the same electricity maps type.

### Double check

- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: electricitymap/contrib/lib/models/events.py
Comment (unitrium) [inline electricitymap/contrib/lib/models/events.py:160]: Since we are doing OOP, how about bringing some part inside the base Mix class?
I think the part about adding the value can be shared.

File: electricitymap/contrib/lib/models/events.py
Comment (unitrium) [inline electricitymap/contrib/lib/models/events.py:140]:
```suggestion
            value = 0 if correct_negative_with_zero else None
```

Comment (unitrium) [reply electricitymap/contrib/lib/models/events.py:140]: Might be easier one line

**Review by unitrium — COMMENTED** [review]
Small suggestions, I think we can improve a bit readability :)

File: electricitymap/contrib/lib/models/events.py
Comment (VIKTORVAV99) [reply electricitymap/contrib/lib/models/events.py:140]: Sure thing, I thought we might want to add something else here in the future (like record the value) but it would be trivial to change back.

File: electricitymap/contrib/lib/models/events.py
Comment (VIKTORVAV99) [reply electricitymap/contrib/lib/models/events.py:160]: They already are in the base mix class?
This just overides that so it also corrects for negative values which we don't want to do for storage but I guess we could move it all up and use a flag instead?

File: electricitymap/contrib/lib/models/events.py
Comment (unitrium) [inline electricitymap/contrib/lib/models/events.py:167]:
```suggestion
        super().add_value(mode, value)
```
In that case I suggest doing this and remove the code below :)

File: electricitymap/contrib/lib/models/events.py
Comment (unitrium) [reply electricitymap/contrib/lib/models/events.py:156]: Indeed, I didn't expended the code so I didn't see it was already in the Mix.
Then you can directly go for my suggestion below

File: electricitymap/contrib/lib/models/events.py
Comment (VIKTORVAV99) [reply electricitymap/contrib/lib/models/events.py:167]: Oh I have not really used that approach before, seems useful!

I'll test it!

**Comment by VIKTORVAV99** [comment]
I realize this PR don't have any tests, do you want me to add some specific ones that just test the add value method before we merge this (after we are otherwise happy with the implementation).

**Comment by unitrium** [comment]
Actually yes, that was the last thing I had in mind, a small unit test to demonstrate that it works and make sure we don't do regressions in the future.

**Comment by VIKTORVAV99** [comment]
> Actually yes, that was the last thing I had in mind, a small unit test to demonstrate that it works and make sure we don't do regressions in the future.

I'll add 3 ones for the ProductionMix and 2 for StorageMix, that should cover all scenarios I think. I'll get those done this weekend.

File: pyproject.toml
Comment (VIKTORVAV99) [inline pyproject.toml:109]: Turns out we weren't running any of the new tests 😅

**Review by unitrium — APPROVED** [review]
Great thanks for implementing this!

**Comment by VIKTORVAV99** [comment]
> Great thanks for implementing this!

Whooo! 🥳

**Pull request merged**
